binding-mruby/mrb-ext/marshal.cpp

```diff
@@ -471,10 +472,12 @@ static mrb_value
 read_value(MarshalContext *ctx)
 {
 	mrb_state *mrb = ctx->mrb;
+
 	int8_t type = ctx->readByte();
+
 	mrb_value value;
-	if (mrb->arena_idx > maxArena)
-		maxArena = mrb->arena_idx;
+	if (mrb->gc.arena_idx > maxArena)
+		maxArena = mrb->gc.arena_idx;
```

Comment:
I don't remember the internals of mruby too well so I'll just trust that this does the right thing :)